runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeSession.java

```diff
@@ -108,25 +114,64 @@ void incrementGemFound(int gemID)
 
 	void updateOreFound(int item, int count)
 	{
+		Instant now = Instant.now();
+		Duration timeSinceStart;
+
 		switch (item)
 		{
 			case ItemID.GOLDEN_NUGGET:
 				nuggetsFound += count;
+
+				timeSinceStart = Duration.between(recentPayDirtMined, now);
+				if (!timeSinceStart.isZero())
+				{
+					nuggetsPerHour = (int) ((double) nuggetsFound * (double) HOUR.toMillis() / (double) timeSinceStart.toMillis());
```

Comment:
Maybe move this calculation to a common method since it is duplicated 6 times

Reply:
Yea I can probably make a function for this